📖 Import windows documentation (#93)
adjust/dev-docs: docs/windows/recording/session-parameters.md

# Set up session parameters

If you [register a callback URL](https://help.adjust.com/en/article/recommended-placeholders-callbacks) in the Adjust dashboard, Adjust sends a GET request to your callback URL when the SDK measures a {term}`session`.

You can configure callback parameters to your servers. Once you configure parameters on an event, the SDK appends them to your callback URL. You can use this information to analyze your users' in-app behavior with your BI system.

:::{note}
The Adjust SDK merges session callback parameters with event callback parameters. Event callback parameters take priority over session callback parameters. This means that if you add a parameter key to both an event and a session, the SDK sends the event parameter.
:::

> Add callback parameters to your event by calling the [`AddSessionCallbackParameter` method](#windows-addsessioncallbackparameter-invocation) with **string** key-value arguments. You can add multiple parameters by calling this method multiple times.

Reviewer: This is all framed as setting up session param, but then this says "add to your event"?

:::{include} /windows/reference/Adjust/recording.md
:start-after: AddSessionCallbackParameter snippet
:end-before: Snippet end
:::

You can remove specific session callback parameters if they're no longer required. To do this, pass the parameter key to the [`RemoveSessionCallbackParameter` method](#windows-removesessioncallbackparameter-invocation).

:::{include} /windows/reference/Adjust/recording.md
:start-after: RemoveSessionCallbackParameter snippet
:end-before: Snippet end
:::

You can remove all session parameters if they're no longer required. To do this, call the [`ResetSessionCallbackParameters` method](#windows-resetsessioncallbackparameters-invocation).

:::{include} /windows/reference/Adjust/recording.md
:start-after: ResetSessionCallbackParameters snippet
:end-before: Snippet end
:::

## Session partner parameters

You can send extra information to your network partners by adding [partner parameters](https://help.adjust.com/en/article/advanced-event-setup#receive-custom-data-with-partner-parameters).

> Adjust sends partner parameters to [external partners](https://help.adjust.com/en/article/integrated-partners) you've set up. This information is useful for more granular analysis and retargeting purposes. Adjust's servers forward these parameters once you've set them up and enabled them for a partner.

Reviewer: ```suggestion
Adjust's servers forward partner parameters to any [external partners](https://help.adjust.com/en/article/integrated-partners) you enable them for. This information is useful for more granular analysis and retargeting purposes.
```
